src/foam/nanos/dig/HTTPDigestSink.java

```diff
@@ -73,6 +73,53 @@ public HTTPDigestSink(String url, String bearerToken, DUGDigestConfig dugDigestC
 
   @Override
   public void put(Object obj, Detachable sub) {
+    try {
+      int responseCode = sendRequest(obj);
+
+      if ( responseCode == HttpServletResponse.SC_OK ) return;
+
+      if ( responseCode == HttpServletResponse.SC_BAD_REQUEST ) { // client error
+        if ( obj instanceof EmailWebhook ) {
+          EmailWebhook ewh = (EmailWebhook) obj ;
+          DAO alarmDAO = (DAO) getX().get("alarmDAO");
+          String emailAddress = ewh.getEmail();
+          String name = "Unable to send request information email to : " + emailAddress;
+          Alarm alarm = (Alarm) alarmDAO.find(EQ(Alarm.NAME, name));
+          if ( alarm == null || ! alarm.getIsActive() ) {
+            alarm = new Alarm.Builder(getX())
+              .setName(ewh.getFirstName() + " " + ewh.getLastName())
+              .setSeverity(LogLevel.ERROR)
+              .setIsActive(true)
+              .setReason(AlarmReason.UNSPECIFIED)
+              .setNote(emailAddress + " user did not receive the email")
```

Comment:
If you are finding the previous alarm, then you can add the email address to the notes.

Reply:
@hchoura please review the suggestion

Reply:
@jlhughes Isn't the email address in the note already?

Reply:
sorry, 'append' to the note. So if there are a number of failures, you'll capture them all in the notes.   
So rather than have the email in the alarm name, and have 30 alarms, we just have one alarm when webhooks fail, and the specific failures are in the note.   
If the failure is common, and you are often having to find and resend, then we should manage webhooks elsewhere, not in alarms.  The alarm should simply get our attention.